src/base/sparsity_pattern.C

```diff
@@ -112,10 +112,28 @@ void Build::handle_vi_vj(const std::vector<dof_id_type> & element_dofs_i,
   const unsigned int n_dofs_on_element_j =
     cast_int<unsigned int>(element_dofs_j.size());
 
+  // It only makes sense to compute hashes and see if we can skip
+  // doing work when there are a "large" amount of DOFs for a given
+  // element. The cutoff for "large" is somewhat arbitrarily chosen
+  // based on a test case with a spider node that resulted in O(10^3)
+  // entries in element_dofs_i for O(10^3) elements. Making this
+  // number larger will disable the hashing optimization in more
+  // cases.
+  bool dofs_seen = false;
+  if (n_dofs_on_element_j > 0 && n_dofs_on_element_i > 256)
+    {
+      auto hash_i = Utility::hashword(element_dofs_i);
+      auto hash_j = Utility::hashword(element_dofs_j);
+      auto final_hash = Utility::hashword2(hash_i, hash_j);
+      auto result = hashed_dof_sets.insert(final_hash);
+      // if insert failed, we have already seen these dofs
```

Comment:
... or we've seen something that hashes to the same value.  I'm sure that only happens one in 2^64 times or whatever but it still scares me a little.  What happens if there **is** a hash collision here?

Reply:
You're right, if there's a hash collision, then we would skip setting some sparsity entries that we would otherwise need. If we store more information, we might be able to make a more thorough check in `DEBUG` mode? Since performance is of the essence here, I was hoping to avoid anything too heavy...

Reply:
Hmmm... I was originally trying to avoid a DEBUG-mode-only check, but I guess if we have an inadequate sparsity pattern then PETSc will scream at us and die even in opt mode?  That's fine then.  I just don't want any possibility of *wrong* results.

Reply:
(And yes, I'm aware that cosmic rays strike more often than one in N_elem/2^64 runs.  But that's the hardware guys' problem.)